### PR TITLE
fix(web): filter out `package.json` from file select

### PR DESCRIPTION
**Pull request by sripwoud**

### CODE REVIEW

File: packages/semaphore/semaphore-12.json
Comment (sripwoud) [inline packages/semaphore/semaphore-12.json:1]: unrelated to this PR, just a format change

File: apps/web/src/hooks/useFiles.ts [hunk 1]
```diff
@@ -20,7 +20,9 @@ export function useFiles() {
     queryFn: async () => fetchFiles(`@zk-kit/${selectedProject}-artifacts@${selectedVersion}`),
     select: data => {
       const regex = new RegExp(`^packages/${selectedProject}/.*\\.(${artifacts.join('|')})$`)
-      return data.filter(file => regex.test(file)).map(file => file.replace(`packages/${selectedProject}/`, ''))
+      return data.filter(file => regex.test(file)).filter(file => !file.endsWith('package.json')).map(file =>
```
Comment (sripwoud) [inline apps/web/src/hooks/useFiles.ts:23]: it would probably be more efficient to edit the regex instead.
But I am bad at regexes, and find them hard to read. So using a second more straightforward `.filter` instead

File: packages/semaphore/semaphore-12.json
Comment (cedoor) [inline packages/semaphore/semaphore-12.json:1]: This should not part of this PR right?

Comment (sripwoud) [reply packages/semaphore/semaphore-12.json:1]: not really
i had left a comment here https://github.com/privacy-scaling-explorations/snark-artifacts/pull/124#discussion_r1683928481
otherwise the ci check will fail

File: apps/web/src/containers/ProjectSelect.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// @ts-expect-error FIXME
```
Comment (sripwoud) [inline apps/web/src/containers/ProjectSelect.tsx:1]: I don't like this.
This is a quick and dirty fix for https://github.com/privacy-scaling-explorations/snark-artifacts/actions/runs/10004163116/job/27652343423?pr=124#step:11:14
(I don't really know why this started happening there)

File: packages/semaphore/semaphore-12.json
Comment (sripwoud) [inline packages/semaphore/semaphore-12.json:1]: unrelated to this PR
but required to have the ci check pass

Comment (sripwoud) [inline packages/semaphore/semaphore-12.json:1]: unrelated to this PR
but required to have the ci check pass

**Review by cedoor — APPROVED** [review]
